### PR TITLE
Replace session key with secret key

### PR DESCRIPTION
**Pull request by antonpirker**
Hi Sergio!

Here an updated pull request for my little refactoring for renaming session_key to secret_key. 
This should be easily mergable!

Hope this helps,
Anton

Ps: Keep up your great work!

### CODE REVIEW

**Comment by SergioBenitez** [comment]
@antonpirker Thanks so much! Can you take a look at the build failure?

File: lib/src/config/custom_values.rs [hunk 1]
```diff
@@ -7,24 +7,24 @@ use config::{Result, Config, Value, ConfigError};
...
 }
 
-impl SessionKey {
+impl SecretKey {
     #[inline(always)]
     pub fn kind(&self) -> &'static str {
         match *self {
-            SessionKey::Generated(_) => "generated",
-            SessionKey::Provided(_) => "provided",
+            secretKey::Generated(_) => "generated",
```
Comment (sganz) [inline lib/src/config/custom_values.rs]: Looks like `secretKey` should be `SecretKey`. Just poking around and spotted this.

**Comment by antonpirker** [comment]
Ok, everything fixed! :)

**Comment by SergioBenitez** [comment]
Awesome! Thanks so much, @antonpirker! Also to you, @sganz.

Merged in 781477fff1d1983897b35dad49c74e925185289c.

**Pull request closed without merging**